### ISSUE TITLE
crossover_rate in OptimizerConfig is accepted but never used by ga()

### ISSUE THREAD

**Issue by ipochi** [jlm429/pyperch#16]
OptimizerConfig defines crossover_rate: float = 0.5 ([config/schema.py#L31](https://github.com/jlm429/pyperch/blob/master/pyperch/config/schema.py#L31)), but the ga() function signature does not accept this parameter ([optim/ga.py#L24-L31](https://github.com/jlm429/pyperch/blob/master/pyperch/optim/ga.py#L24-L31)):

```
def ga(
    params: List[torch.Tensor],
    random: np.random.Generator,
    closure: Closure,
    population_size: int = 50,
    mutation_rate: float = 0.1,
    step_size: float = 0.1,
) -> float:
```
The crossover behavior is hardcoded to uniform 50/50 in _crossover() ([optim/ga.py#L192](https://github.com/jlm429/pyperch/blob/master/pyperch/optim/ga.py#L192)):

```
mask = torch.bernoulli(torch.full_like(p1, 0.5))
```

A user can set crossover_rate to any value in OptimizerConfig without error or warning, but it has no effect on the GA behavior.

Expected Behavior
Either:

ga() should accept `crossover_rate` and use it (e.g., to control the fraction of the population produced via crossover, or the per-element crossover probability in the mask), or `crossover_rate` should be removed from OptimizerConfig to avoid misleading users

Steps to Reproduce
```
import torch, numpy as np
from pyperch.config.schema import OptimizerConfig
from pyperch.optim.ga import ga

torch.manual_seed(42)
param = torch.randn(5, requires_grad=True).clone().detach()

def run_ga(crossover_rate_value):
    p = param.clone().requires_grad_(True)
    torch.manual_seed(42)
    rng = np.random.default_rng(42)
    closure = lambda: float((p ** 2).sum())
    config = OptimizerConfig(name="ga", crossover_rate=crossover_rate_value)
    loss = ga(params=[p], random=rng, closure=closure,
              population_size=10, mutation_rate=0.1, step_size=0.1)
    return loss, p.clone().detach()

loss1, w1 = run_ga(0.1)
loss2, w2 = run_ga(0.9)

print(f"crossover_rate=0.1 → loss={loss1:.6f}")
print(f"crossover_rate=0.9 → loss={loss2:.6f}")
print(f"Weights identical: {torch.allclose(w1, w2)}")
```

Expected Output

Different crossover_rate values should produce different GA behavior (different losses and/or weights).

Actual Output
```
crossover_rate=0.1 → loss=1.122053
crossover_rate=0.9 → loss=1.122053
Weights identical: True
```

Both runs produce bitwise-identical results. `crossover_rate` is stored in OptimizerConfig but never passed to or read by ga().

Environment
pyperch version: 0.2.2
Python: 3.12

**Comment by jlm429** [comment]
Closing - v. 3.0+ optimizers replacing 0.2.x.
